### PR TITLE
DAOS-3835 control: Convert DAOS err codes to strings

### PR DESCRIPTION
**Pull request by mjmac**
Use the d_errstr() function in libgurt to provide
somewhat-more comprehensible error messages to control
API callers. Uses a server-side gRPC interceptor to
handle the conversion before sending the response.

### CODE REVIEW

File: src/control/drpc/status.go [hunk 1]
```diff
@@ -25,130 +25,145 @@
...
+	dErrStr := C.GoString(C.d_errstr(C.int(ds)))
+	return fmt.Sprintf("DAOS error (%d): %s", ds, dErrStr)
+}
+
 const (
 	// DaosSuccess indicates no error
 	DaosSuccess DaosStatus = 0
 	// DaosNoPermission indicates that access to a resource was denied
-	DaosNoPermission = -C.DER_NO_PERM
+	DaosNoPermission DaosStatus = -C.DER_NO_PERM
```
Comment (kjacque) [inline src/control/drpc/status.go]: For some reason I thought the type was implied by the first item in the list in these const lists. Is that only when using iota?

File: src/control/server/interceptors.go [hunk 2]
```diff
@@ -133,3 +134,31 @@ func streamErrorInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.St
...
+
+// dErrFromStatus converts a numeric DAOS return status code
+// into an error.
+func dErrFromStatus(sg statusGetter) error {
+	dStatus := sg.GetStatus()
+	if dStatus == 0 {
+		return nil
+	}
+
+	return drpc.DaosStatus(-dStatus)
```
Comment (kjacque) [inline src/control/server/interceptors.go]: The status coming back from the ioserver is already negative, isn't t?

File: src/control/drpc/status.go (hunk 1 above)
Comment (mjmac) [reply src/control/drpc/status.go]: Yeah, with iota you set the type in the first assignment and then it's carried through to the other implicit values.

File: src/control/server/interceptors.go (hunk 2 above)
Comment (mjmac) [reply src/control/server/interceptors.go]: Oops, yeah. Fixed. For some reason I was thinking the response status was a uint32. Interestingly, d_errstr() gives the same results regardless of the sign. I'll add a quick test case for this helper, though.

**Comment by mjmac** [comment]
Known test failures:
- DAOS-4410
- DAOS-4285

**Pull request merged**